.github/CONTRIBUTING.md

```diff
@@ -4,67 +4,126 @@ Water.css becomes better for everyone when people like you help make it better!
+Also, everyone is welcome to contribute on issues [tagged with **`help wanted`**, you can find it filtered here](https://github.com/kognise/water.css/issues?q=is%3Aopen+is%3Aissue+label%3A%22help+wanted%22).
+
+## Add or change styles
 
+There are a few rules for working in the Water.css source code:
 
-## How to make a pull request
+1. The styles must not use any classes or IDs. Water.css is made entirely of basic style rules that only target HTML elements.
 
-It is a few general rules of thumb about making pull requests:
+2. Don't specify colors directly in the source files (`src/parts/*`). Only use named variables there, which are declared in `src/variables-*.css`.
+
+  ❌ Bad:
+  ```css
+  color: #fff;
+  ```
+
+  ✔ Good:
+  ```css
+  color: var(--text-bright);
+  ```
+
+3. Reuse existing colors where possible. Before introducing a new color to our palette, check if one of the existing colors fits your needs.
+
+4. If you introduce a new variable, make sure to declare it both in `variables-dark.css` and `variables-light.css` so it's available in both themes.
+
+## Make your pull request
+
+Once you're happy with your changes, you need to **commit** them, create a **changelog** and **submit a pull request**.
+
+A few general rules of thumb about what makes a good pull request:
 
 - Make sure that your pull request covers a small and well defined scope
 - Make small commits with clear and explainful messages
-- You need to provide a clear description about your contribution on GitHub
+- Provide a clear description about your contribution on GitHub
+
+When you commit code, Water.css will run some checks to make sure that your changes have no errors and match the project's coding style – a process called **Linting**. It will also verify that all **colors are accessible**, which means they need to have enough contrast to be easily readable.
 
-Before making a pull request, make sure you lint your code and generate a changelog.
+If the code you introduced has errors or some of the colors aren't accessible, you'll see an error and the commit will abort.
+You can check whether your changes pass all tests before committing them by running:
 
-**Linting**
+```
+yarn validate
+```
 
-If you aren't familiar, a linter is a program that automatically checks your code for formatting issues and potential bugs. Our linters are configured to automatically fix what they can so you can typically run one command and be done with it:
+If there are errors, you can try to fix them automatically by running:
 
 ```
 yarn lint
 ```
 
-**Changelogs**
+<details>
+  <summary>ℹ A note about puppeteer and WSL</summary>
+  <br>
+  <blockquote>
+    The accessibility checks use puppeteer, a tool that uses Chrome to render websites "headlessly", without a visible interface. In some environments like the <a href="https://aka.ms/wsl">Windows Subsystem for Linux</a>, you'll need to manually configure and run an X-Server in order for puppeteer to work.
```

Comment:
Ooh, I didn't realize this!

Reply:
Yeah, right now you need to run VcXsrv or a similar X-Server in Windows and point the `DISPLAY` env variable in the Linux environment at it.

But Microsoft announced at Build a couple days ago that graphical apps and GPU access is coming to WSL, so I assume they'll add a native X-Server integration then :)